### DISCUSSION TITLE
[WASM/Android] Frame.Navigate laggy

### DISCUSSION THREAD

**Discussion by emceelovin** [unoplatform/uno#8216] [Q&A]
When using Frame.Navigate to navigate to Page's, there is a huge lag on WASM and Android. Has anyone else experienced that? It's fine on Windows Package build.

**Comment by jeromelaban** [comment]
Thanks for the report. Can you explain with more details what "huge lag" means?

**Reply by emceelovin** [reply]
It takes a significant amount of time to navigate to the page supplied to Frame.Navigate(typeof(PageType)). I narrowed it down to Frame.Navigate inside of my NavigationView OnSelectionChanged. Switching between items in NavigationView is decently fast. When I add the Frame.Navigate(typeof(PageType)) back, it takes a solid 1-2 seconds plus to load a Page.

**Reply by jeromelaban** [reply]
Thanks. Are you able to extract a repro sample so we're sure we're talking about the same issue? You can follow the steps [here](https://github.com/unoplatform/uno/blob/master/.github/ISSUE_TEMPLATE/bug-report.yml) (The "how to reproduce it" section).

**Reply by emceelovin** [reply]
I'll make a simple project as soon as I can. The code is very small I could probably even paste it here. I'm going to try and do that real quick.
